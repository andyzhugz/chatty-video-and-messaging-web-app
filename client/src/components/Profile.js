import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { changePassword } from '../javascripts/authRequests'
import { getUser } from '../javascripts/contact';
import '../stylesheets/Profile.css';
import Avatar from '../images/AvatarCat.png';
import NaviBar from './NaviBar';

const Profile = () => {
  const [username, setUsername] = useState('');
  const [userData, setUserData] = useState({name:'', registration:''});
  const [error, setError] = useState(false);
  let history = useHistory();

  
  const validation = (password) => {
    let pattern = new RegExp(/^[0-9a-zA-Z]{8,}$/);
    if (!pattern.test(password)) {
        const msg = "At least 8 alphanumeric characters for password";
        setError(msg); 
        return false;
      }
    return true;
  };

  const handleChangePassword = (event) => {
    event.preventDefault();
    let email = username;
    let newPassword = document.getElementById('newPassword').value;
    if (validation(newPassword)) {
      changePassword(email, newPassword).then((res) => {
        console.log(res);
        if (res.status === 200) {
          localStorage.clear();
          history.push("/login");
        }
        else {
          setError("Failed to change password");
        }
      }).catch((e) => {
        setError("Failed to change password");
        console.log(e);
      });
    }

  }

  useEffect(() => {
    const loggedInUser = localStorage.getItem("curr_user");
    if (loggedInUser && loggedInUser !== "") {
      setUsername(loggedInUser);
      getUser(loggedInUser)
      .then((res) => {
        const registrationDate = res.data.registrationDate;
        setUserData({name: loggedInUser, registration: registrationDate})
      });
    }
    else {
      history.push("/login");
    }
  }, []);

  return (
    <div>
      <NaviBar />
      <h1 align="center">Profile</h1>
      <div className="mainProfile">
        <img src={Avatar} alt="Girl in a jacket" className="avatar" />
        <p id="userName" className="data" align="center">
          {userData.name}
        </p>
        <p id="registrationDate" className="data" align="center">
          {userData.registration}
        </p>
        <form className="formLogin">
          <input
            id="newPassword"
            className="newPassword"
            type="password"
            align="center"
            placeholder="new password"
          />
          <button id="resetBtn" className="resetBtn" align="center" onClick={handleChangePassword}>
            <b>Change Password</b>
          </button>
          <button className="deactivateBtn" align="center">
            <b>Deactivate</b>
          </button>
        </form>
        â€‹{error && <><div id="error" style={{ color: 'red', fontSize: 12, textAlign: "center" }}>{error}</div><br /></>}<br />
      </div>
    </div>
  );
};

export default Profile;
