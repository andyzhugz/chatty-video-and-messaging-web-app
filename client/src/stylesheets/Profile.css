body {
  background-image: url('../images/Earth.png');
  background-repeat: no-repeat;
  background-position: center;
  font-family: 'Ubuntu', sans-serif;
}

h1 {
  margin-top: 100px;
  color: #fff;
}

.mainProfile {
  background-color: #ffffff;
  width: 600px;
  height: 540px;
  margin: 3em auto;
  border-radius: 1.5em;
  box-shadow: 0px 11px 35px 2px rgba(0, 0, 0, 0.14);
}

.avatar {
  width: 200px;
  height: 200px;
  margin: 25px 200px;
  border-radius: 1.5em;
  box-shadow: 0px 11px 35px 2px rgba(0, 0, 0, 0.14);
}

.data {
  color: #8c55aa;
  font-family: 'Ubuntu', sans-serif;
  font-weight: bold;
  font-size: 23px;
}

.newPassword {
  width: 40%;
  color: rgb(38, 50, 56);
  font-weight: 700;
  font-size: 14px;
  letter-spacing: 1px;
  background: rgba(136, 126, 126, 0.04);
  padding: 10px 20px;
  border: none;
  border-radius: 20px;
  outline: none;
  box-sizing: border-box;
  border: 2px solid rgba(0, 0, 0, 0.02);
  margin-bottom: 50px;
  margin-left: 7%;
  text-align: center;
  margin-bottom: 27px;
  font-family: 'Ubuntu', sans-serif;
}

.resetBtn {
  cursor: pointer;
  border-radius: 5em;
  color: #fff;
  background: linear-gradient(to right, #9c27b0, #e040fb);
  border: 0;
  padding-left: 40px;
  padding-right: 40px;
  padding-bottom: 10px;
  padding-top: 10px;
  font-family: 'Ubuntu', sans-serif;
  font-weight: 700;
  margin-left: 7%;
  font-size: 13px;
  box-shadow: 0 0 20px 1px rgba(0, 0, 0, 0.04);
  display: inline;
}

.deactivateBtn {
  color: red;
  cursor: pointer;
  border-radius: 5em;
  background: linear-gradient(to right, #938e94, #1d171f);
  border: 0;
  padding-left: 40px;
  padding-right: 40px;
  padding-bottom: 10px;
  padding-top: 10px;
  font-family: 'Ubuntu', sans-serif;
  font-weight: 700;
  margin-left: 39%;
  font-size: 13px;
  box-shadow: 0 0 20px 1px rgba(0, 0, 0, 0.04);
  display: inline;
}

form.formLogin {
  padding-top: 40px;
}

a {
  text-shadow: 0px 0px 3px rgba(117, 117, 117, 0.12);
  color: #e1bee7;
  text-decoration: none;
}

@media (max-width: 600px) {
  .main {
    border-radius: 0px;
  }
}
