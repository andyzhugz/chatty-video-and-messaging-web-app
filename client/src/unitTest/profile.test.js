import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import Profile from '../components/Profile.js';

const renderer = require('react-test-renderer');

// snapshot testings
describe('Test snapshot', () => {
  test('Test Profile', () => {
    const component = renderer.create(<Profile />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// functonal component functions tests
describe('Test functonal component functions', () => {
  let container;
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  test('Test function inside Login component', () => {
    act(() => {
      ReactDOM.render(<Profile />, container);
    });
  });

});
