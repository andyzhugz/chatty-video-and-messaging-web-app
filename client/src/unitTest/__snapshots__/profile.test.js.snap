// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Test Login 1`] = `
<div>
  <h1
    align="center"
  >
    Welcome to Our App
  </h1>
  <div
    className="main"
  >
    <p
      align="center"
      className="sign"
    >
      Login
    </p>
    <form
      className="formLogin"
    >
      <input
        align="center"
        className="userName"
        id="userName"
        placeholder="Username: cat@gmail.com"
        type="text"
      />
      <input
        align="center"
        className="pass"
        id="password"
        placeholder="Password"
        type="password"
      />
      <p
        align="center"
        className="forgotPwd"
      >
        <a
          href="#"
        />
        Forgot Password?
      </p>
      <input
        className="loginBtn"
        disabled={false}
        id="loginBtn"
        onClick={[Function]}
        type="button"
        value="Sign In"
      />
      <a
        href="/Register"
        onClick={[Function]}
      >
        <button
          align="center"
          className="registerBtn"
          id="registerBtn"
        >
          <b>
            Register
          </b>
        </button>
      </a>
      <br />
    </form>
  </div>
</div>
`;

exports[`Test snapshot Test Profile 1`] = `
<div>
  <div>
    <ul
      id="nav_ul"
    >
      <li
        className="nav_li"
      >
        <a
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        className="nav_li"
      >
        <a
          href="/main"
        >
          Messaging
        </a>
      </li>
      <li
        className="nav_li"
      >
        <a
          href="/status"
        >
          Status
        </a>
      </li>
    </ul>
  </div>
  <h1
    align="center"
  >
    Profile
  </h1>
  <div
    className="mainProfile"
  >
    <img
      alt="Girl in a jacket"
      className="avatar"
      src="AvatarCat.png"
    />
    <p
      align="center"
      className="data"
      id="userName"
    >
       
      data.name
       
    </p>
    <p
      align="center"
      className="data"
      id="registrationDate"
    >
       
      data.registrationDate
       
    </p>
    <form
      className="formLogin"
    >
      <input
        align="center"
        className="newPassword"
        id="newPassword"
        placeholder="new password"
        type="password"
      />
      <button
        align="center"
        className="resetBtn"
        id="resetBtn"
      >
        <b>
          Change Password
        </b>
      </button>
      <button
        align="center"
        className="deactivateBtn"
      >
        <b>
          Deactivate
        </b>
      </button>
    </form>
    â€‹
  </div>
</div>
`;
